docs/src/APIs/Soil.md

```diff
@@ -17,6 +17,71 @@ ClimaLand.Soil.RichardsParameters
 ClimaLand.Soil.EnergyHydrologyParameters
 ```
 
+## Soil Constructor Methods
+
```

Comment:
We can also add:
https://github.com/CliMA/ClimaLand.jl/blob/6e43b5c3c691bd21802441d88bb886a0b4ed5045/src/standalone/Soil/rre.jl#L81

and

https://github.com/CliMA/ClimaLand.jl/blob/6e43b5c3c691bd21802441d88bb886a0b4ed5045/src/standalone/Soil/energy_hydrology.jl#L281